Review Charm for Listing

kelkawi-a opened the pull request:
PR for reviewing charm code as defined in: https://discourse.charmhub.io/t/reviewing-charms/11698.

phvalguima reviewed (commented):
It looks good! I left some notes that are more related to style.

I am missing the lib for `airbyte-server` relation.

kelkawi-a commented on src/charm.py:
```diff
@@ -0,0 +1,221 @@
+        super().__init__(*args)
+        self._state = State(self.app, lambda: self.model.get_relation("peer"))
+
+        self.name = "airbyte-webapp"
```
`self.app.name` refers to the name the application is deployed with (by default that is `airbyte-ui-k8s`). `self.name` here refers to the name of the container, which has parity with the name of the OCI image `airbyte/webapp`.

kelkawi-a commented on src/charm.py:
```diff
@@ -0,0 +1,221 @@
+            "summary": "airbyte layer",
+            "services": {
+                self.name: {
+                    "summary": self.name,
```
`self.app.name` can vary depending on the name of your deployment, using `self.name` makes this a uniform value.

kelkawi-a commented on src/relations/airbyte_server.py:
```diff
@@ -0,0 +1,65 @@
+        self.charm._state.airbyte_server = {
+            "name": event.relation.data[event.app].get("server_name"),
+            "status": event.relation.data[event.app].get("server_status"),
+        }
```
Doing this causes issue with the `state` manager setup because pydantic models are not JSON serializable. There is also no lib for this relation as it only exists between these two charms, so no lib is required at this point.

kelkawi-a replied:
I could do something like the following if suitable:
```python
 self.charm._state.airbyte_server = AirbyteServerRelationModel(
       name=event.relation.data[event.app].get("server_name"),
       status=event.relation.data[event.app].get("server_status"),
 ).model_dump()
```

kelkawi-a commented:
> It looks good! I left some notes that are more related to style.
> 
> I am missing the lib for `airbyte-server` relation.

Thanks for the review! I've addressed most of the comments in #11
